block_engine/src/block_engine.rs

```diff
@@ -616,33 +629,32 @@ impl BlockEngineRelayerHandler {
         accounts_of_interest: &mut TimedCache<Pubkey, u8>,
         programs_of_interest: &mut TimedCache<Pubkey, u8>,
         address_lookup_table_cache: &DashMap<Pubkey, AddressLookupTableAccount>,
+        ofac_addresses: &HashSet<Pubkey>,
     ) -> Option<ExpiringPacketBatch> {
-        let filtered_packets: Vec<ProtoPacket> = block_engine_batches
-            .packet_batches
-            .into_iter()
-            .flat_map(|b| {
-                b.iter()
-                    .filter_map(|p| {
-                        let tx: VersionedTransaction = p.deserialize_slice(..).ok()?;
-
-                        let is_forwardable =
-                            is_aoi_in_static_keys(&tx, accounts_of_interest, programs_of_interest)
-                                || is_aoi_in_lookup_table(
-                                    &tx,
-                                    accounts_of_interest,
-                                    programs_of_interest,
-                                    address_lookup_table_cache,
-                                );
-
-                        if is_forwardable {
-                            Some(packet_to_proto_packet(p)?)
-                        } else {
-                            None
+        let mut filtered_packets = Vec::new();
```

Comment:
could use size hint to avoid realloc+copy

Reply:
true that